fluss-client/src/main/java/com/alibaba/fluss/client/write/Sender.java

```diff
@@ -172,7 +172,9 @@ public void run() {
     public void runOnce() throws Exception {
         if (idempotenceManager.idempotenceEnabled()) {
             // may be wait for writer id.
-            idempotenceManager.maybeWaitForWriterId();
+            Set<PhysicalTablePath> physicalTablePaths =
+                    accumulator.getPhysicalTablePathsInBatches();
+            idempotenceManager.maybeWaitForWriterId(physicalTablePaths);
```

Comment:
We should only request the writer_id when there are actual table paths in the batch. This helps avoid unnecessary requests when the writer/sender is running but no data has been appended.